app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -447,29 +447,41 @@ export default SiteHeaderComponent.extend({
 
   init() {
     this._super(...arguments);
-
     this._resizeObserver = null;
   },
 
+  updateHeaderOffset() {
+    // check offset from top of the page, in case there's a custom header above
+    // and adjust on scroll in case it's not sticky
+    if (this.scrollAnimationFrame) {
+      cancelAnimationFrame(this.scrollAnimationFrame);
+    }
+
+    this.scrollAnimationFrame = requestAnimationFrame(() => {
+      const currentHeaderWrapTop = this.headerWrap.getBoundingClientRect().top;
+
+      document.documentElement.style.setProperty(
+        "--header-offset",
+        `${this.headerWrap.offsetHeight + currentHeaderWrapTop}px`
+      );
+    });
+  },
+
   didInsertElement() {
     this._super(...arguments);
 
     this.appEvents.on("site-header:force-refresh", this, "queueRerender");
 
     this.headerWrap = document.querySelector(".d-header-wrap");
+
     if (this.headerWrap) {
       schedule("afterRender", () => {
         this.header = this.headerWrap.querySelector("header.d-header");
-        const headerOffset = this.headerWrap.offsetHeight;
-        const headerTop = this.header.offsetTop;
-        document.documentElement.style.setProperty(
-          "--header-offset",
-          `${headerOffset}px`
-        );
-        document.documentElement.style.setProperty(
-          "--header-top",
```

Comment:
We're still using `--header-top` in some core stylesheets, so I don't think we can remove it without refactoring those places